python/Ganga/Core/InternalServices/ShutdownManager.py

```diff
@@ -38,157 +38,110 @@
-        """
-        # extract the priority number from the function element
-        p1 = f1[0][0]
-        p2 = f2[0][0]
-        # sort in reversed order
-        return cmp(p2, p1)
-
-    def add_priority(x):
-        """
-        add a default priority to the functions not defining one (default priority=sys.maxint)
-        return a list containg ((priority,func),*targs,*kargs) elements
-        """
-        import sys
-        func = x[0]
-        if isinstance(func, tuple) and len(x[0]) == 2:
-            return x
-        else:
-            new = [(sys.maxsize, func)]
-            new.extend(x[1:])
-            return new
-
-    atexit._exithandlers = map(add_priority, atexit._exithandlers)
-    atexit._exithandlers.sort(priority_cmp)
-
-    logger.info("Stopping Job processing before shutting down Repositories")
-
-    from Ganga.Core.GangaThread.WorkerThreads import _global_queues
-    if _global_queues:
-        _global_queues.freeze()
-
-    import inspect
-    while atexit._exithandlers:
-
-        (priority, func), targs, kargs = atexit._exithandlers.pop()
         try:
-            if hasattr(func, 'im_class'):
-                for cls in inspect.getmro(func.__self__.__class__):
-                    if getName(func) in cls.__dict__:
-                        logger.debug(getName(cls) + " : " + getName(func))
-                func(*targs, **kargs)
-            else:
-                logger.debug("noclass : " + getName(func))
-                #print("%s" % str(func))
-                #print("%s" % str(inspect.getsourcefile(func)))
-                #func(*targs, **kargs)
-                ## This attempts to check for and remove externally defined shutdown functions which may interfere with the next steps!
-                if str(inspect.getsourcefile(func)).find('Ganga') != -1:
-                    func(*targs, **kargs)
+            getStackTrace()
+            if monitoring_component.alive:
+                monitoring_component.disableMonitoring()
+                monitoring_component.stop()
+                monitoring_component.join()
         except Exception as err:
-            s = 'Cannot run one of the exit handlers: %s ... Cause: %s' % (getName(func), str(err))
-            logger.debug(s)
-            try:
-                import os
-                logger.debug("%s" % os.path.join(os.path.dirname(os.path.abspath(inspect.getfile(func)))) )
-                logger.debug("\n%s" % inspect.getsource(func))
-            except Exception as err2:
-                logger.debug("Error getting source code and failure reason: %s" % str(err2))
-
-    from Ganga.Core.GangaThread.WorkerThreads import shutDownQueues
-    shutDownQueues()
-
-    logger.info("Shutting Down Ganga Repositories")
-    from Ganga.Runtime import Repository_runtime
-    Repository_runtime.shutdown()
-
-    from Ganga.Core.InternalServices import Coordinator
+            logger.warning("Exception raised while stopping the monitoring: %s" % err)
```

Comment:
For more detailed information (e.g. the full stacktrace), perhaps these should use [`logger.exception()`](https://docs.python.org/2.6/library/logging.html#logging.Logger.exception)?